addons/sale_project/tests/test_sale_project.py

```diff
@@ -1304,3 +1305,52 @@ def test_so_with_service_product_negative_qty(self):
         so.action_confirm()
         self.assertFalse(self.product_order_service2.project_id.task_ids)
         self.assertFalse(sol.task_id)
+
+    def test_sale_order_lines_associated_sale_orders_of_parent_and_child_partners(self):
+        """
+        Test Case Steps:
+        1. Create a child partner linked to the parent partner.
+        2. Create and confirm a sale order for the parent partner.
+        3. Create and confirm a sale order for the child partner.
+        4. Set the parent partner on the project and fetch the sale order lines.
+        5. Verify that child partner's sale order lines are returned.
+        6. Set the child partner on the project and fetch the sale order lines.
+        7. Verify parent partner's sale order lines are returned.
+        """
+        child_partner = self.env['res.partner'].create({
+            'name': 'Child Partner',
+            'parent_id': self.partner.id,
+        })
+        sale_order_1, sale_order_2 = self.env['sale.order'].create([
+            {
+                'partner_id': self.partner.id,
+                'order_line': [
+                    Command.create({
+                        'product_id': self.product_order_service1.id,
+                        'product_uom_qty': 10,
+                    })
+                ]
+            },
+            {
+                'partner_id': child_partner.id,
+                'order_line': [
+                    Command.create({
+                        'product_id': self.product_order_service2.id,
+                        'product_uom_qty': 5,
+                    })
+                ]
+            }
+        ])
+        (sale_order_1 + sale_order_2).action_confirm()
```

Comment:
fyi, we usually use the following syntax when we want to merge a recordset instead of the '+'
```py
recordset = sale_order_1 | sale_order_2
```